mods/ra/maps/allies-08a/allies08a.lua

```diff
@@ -0,0 +1,132 @@
+		units[unit] = cells[i]
+	end
+	Chronosphere.Chronoshift(units)
+
+	Trigger.AfterDelay(DateTime.Seconds(3), function()
+		greece.MarkCompletedObjective(DefendChronosphere)
+		greece.MarkCompletedObjective(KeepBasePowered)
+	end)
+end
+
+PowerFailure = function()
+	greece.MarkFailedObjective(KeepBasePowered)
+end
+
+MissionFailed = function()
+	greece.MarkFailedObjective(DefendChronosphere)
+end
+
+ticked = TimerTicks
+Tick = function()
+	ussr.Cash = 5000
+
+	if ussr.HasNoRequiredUnits() then
+		greece.MarkCompletedObjective(DefendChronosphere)
+		greece.MarkCompletedObjective(KeepBasePowered)
+	end
+
+	if greece.HasNoRequiredUnits() then
+		ussr.MarkCompletedObjective(BeatAllies)
+	end
+
+	if ticked > 0 then
+		UserInterface.SetMissionText("Chronosphere experiment completes in " .. Utils.FormatTime(ticked), TimerColor)
+		ticked = ticked - 1
+	elseif ticked == 0 and (greece.PowerState ~= "Normal") then
+		PowerFailure()
+	elseif ticked == 0 then
+		DefendChronosphereCompleted()
+		ticked = ticked - 1
+	end
+end
+
+WorldLoaded = function()
+	greece = Player.GetPlayer("Greece")
+	ussr = Player.GetPlayer("USSR")
+	germany = Player.GetPlayer("Germany")
+
+	DefendChronosphere = greece.AddPrimaryObjective("Defend the Chronosphere and the Tech Center \nat all costs.")
+	KeepBasePowered = greece.AddPrimaryObjective("The Chronosphere must have power when the \ntimer runs out.")
+	BeatAllies = ussr.AddPrimaryObjective("Defeat the Allied forces.")
+
+	Trigger.OnObjectiveCompleted(greece, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(greece, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerLost(greece, function()
+		Trigger.AfterDelay(DateTime.Seconds(1), function()
+			Media.PlaySpeechNotification(greece, "MissionFailed")
+		end)
+	end)
+	Trigger.OnPlayerWon(greece, function()
+		Trigger.AfterDelay(DateTime.Seconds(1), function()
+			Media.PlaySpeechNotification(greece, "MissionAccomplished")
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(7), function()
+		Media.PlaySpeechNotification(greece, "TwentyMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(17), function()
+		Media.PlaySpeechNotification(greece, "TenMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(22), function()
+		Media.PlaySpeechNotification(greece, "WarningFiveMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(24), function()
+		Media.PlaySpeechNotification(greece, "WarningThreeMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(26), function()
+		Media.PlaySpeechNotification(greece, "WarningOneMinuteRemaining")
+	end)
+
+	PowerProxy = Actor.Create("powerproxy.paratroopers", false, { Owner = ussr })
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+	TimerColor = greece.Color
+
+	Trigger.OnAnyKilled(ObjectiveBuildings, MissionFailed)
```

Comment:
Another minor nit: I'd prefer not adding too many one-line functions:
`Trigger.OnAnyKilled(ObjectiveBuildings, function() greece.MarkFailedObjective(DefendChronosphere) end)`